test: add code snippets for loading data from BigQuery Job

ashleyxuu opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

ashleyxuu committed:
test: add code snippets for loading data from BigQuery Job

tswast commented on samples/snippets/load_data_from_biquery_job_test.py:
```diff
@@ -0,0 +1,37 @@
+    client = bigquery.Client(project=project, location=location)
+
+    # Job ID inserted based on the query results selcted to explore
+    job_id = "a9dbb6a9-db2d-46a5-a497-8cca2159ddeb"
```
This job won't always exist. They get cleaned up every X days (60, I think?)

Maybe run a query before the [START] tag?

ashleyxuu committed:
fix: address the comments

tswast reviewed (approved):
Thanks!

tswast reviewed (changes requested):
Please investigate test failure.

```
=================================== FAILURES ===================================
_____________ test_bigquery_dataframes_load_data_from_bigquery_job _____________

    def test_bigquery_dataframes_load_data_from_bigquery_job():
        import bigframes.pandas as bpd
    
>       bpd.options.bigquery.project = "bigframes-dev"

[samples/snippets/load_data_from_biquery_job_test.py:19](https://cs.corp.google.com/piper///depot/google3/samples/snippets/load_data_from_biquery_job_test.py?l=19): 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <bigframes._config.bigquery_options.BigQueryOptions object at 0x7f000edb9a30>
value = 'bigframes-dev'

    @project.setter
    def project(self, value: Optional[str]):
        if self._session_started and self._project != value:
>           raise ValueError(SESSION_STARTED_MESSAGE.format(attribute="project"))
E           ValueError: Cannot change 'project' once a session has started. Call bigframes.pandas.close_session() first, if you are using the bigframes.pandas API.

[bigframes/_config/bigquery_options.py:100](https://cs.corp.google.com/piper///depot/google3/bigframes/_config/bigquery_options.py?l=100): ValueError
=============================== warnings summary ===============================
```

tswast commented on samples/snippets/load_data_from_biquery_job_test.py:
```diff
@@ -0,0 +1,47 @@
+
+
+def test_bigquery_dataframes_load_data_from_bigquery_job():
+    import bigframes.pandas as bpd
+
+    bpd.options.bigquery.project = "bigframes-dev"
+    bpd.options.bigquery.location = "us"
+
+    query_or_table = "bigquery-public-data.ml_datasets.penguins"
+    df = bpd.read_gbq(query_or_table)
+    df.to_pandas()
+    JOB_ID = df.query_job.job_id
```
This would likely be simpler if we used the BigQuery client library. I understand why we might not want to though because of the duplicate imports and client library construction in the actual code sample. 🤔

ashleyxuu replied:
The failed tests have passed. Do you think it is good to keep it as is?

ashleyxuu committed:
fix: fix the broken test

ashleyxuu committed:
Merge branch 'main' into b307728748-code-snippets-query-results

ashleyxuu committed:
Merge branch 'main' into b307728748-code-snippets-query-results

ashleyxuu committed:
use BigQuery Client library to get the job_id

tswast reviewed (approved):
Love it!

Edit: Oops. Noticed one issue.

tswast commented on samples/snippets/load_data_from_biquery_job_test.py:
```diff
@@ -0,0 +1,50 @@
+    import bigframes.pandas as bpd
+
+    # Project ID inserted based on the query results selected to explore
+    project = "bigframes-dev"
```
Let's make this not reference our project. Define a `your_project_id` variable above.

```suggestion
    project = your_project_id
```

shobsi committed:
feat: Implement operator `@` for `DataFrame.dot` (#139)

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://togithub.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes b/297502513 🦕

ashleyxuu committed:
fix: fix the comments

ashleyxuu committed:
Merge branch 'main' into b307728748-code-snippets-query-results

tswast reviewed (approved):
Love it!
